gnocchi/rest/api.py

```diff
@@ -1484,35 +1579,49 @@ def post(self, metric_id, start=None, stop=None, aggregation='mean',
                 abort(400, "Invalid value for stop")
 
         try:
-            predicate = storage.MeasureQuery(query)
-        except storage.InvalidQuery as e:
+            predicate = self.MeasureQuery(query)
+        except self.MeasureQuery.InvalidQuery as e:
             abort(400, six.text_type(e))
 
         if granularity is not None:
             granularity = sorted(
                 map(utils.to_timespan, arg_to_list(granularity)),
                 reverse=True)
 
-        results = {}
+        metrics_and_aggregations = collections.defaultdict(list)
+
+        for metric in metrics:
+            if granularity is None:
+                granularity = sorted((
+                    d.granularity
+                    for d in metric.archive_policy.definition),
```

Comment:
i imagine we can just check aggregations here while we're looping through the list.

Reply:
I don't understand what you want to check here?

Reply:
Yeah I don't think it matters at that point :)